src/api/java/com/minecolonies/api/crafting/ImmutableItemStorage.java

```diff
@@ -0,0 +1,27 @@
+package com.minecolonies.api.crafting;
+
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Immutable ItemStorage version.
+ */
+public class ImmutableItemStorage extends ItemStorage
+{
+    /**
+     * Creates an instance of the storage.
+     *
+     * @param storage the mutable itemstorage to create it from.
+     */
+    public ImmutableItemStorage(@NotNull final ItemStorage storage)
+    {
+        super(storage.getItemStack(), storage.ignoreDamageValue(), storage.ignoreNBT());
+        super.setAmount(storage.getAmount());
+    }
+
+    @Override
+    public void setAmount(final int amount)
+    {
+        throw new UnsupportedOperationException("Immutable instance of ItemStorage can't set value!");
```

Comment:
So question, could we make this throw an exception in dev, but simply log an error in prod? 
just so we don't get crashes at least during prod if it's somehow missed in dev.

otherwise looks great

Reply:
I actually do want it to crash "for now". If this here happens there would've been data corruption and just logging could potentially result in infinite loops and stackoverflow.

Reply:
To be clear, this is 'corruption' is in-memory only, at least in the scenarios we've seen so far, since it's been on non-persisted ItemStorages. Correct?

Reply:
Yes, the problem is that if we don't increase the count here, but loop over this and use the count in a loop, this would be an infinite loop. I want to prevent that chance.

Reply:
Understood, and agreed.